Classes/BITChannel.m

```diff
@@ -164,15 +191,15 @@ - (void)enqueueTelemetryItem:(BITTelemetryData *)item {
       return;
     }
     
-    // Enqueue item
-    NSDictionary *dict = [self dictionaryForTelemetryData:item];
-    [strongSelf appendDictionaryToJsonStream:dict];
-    
-    if (strongSelf.dataItemCount >= self.maxBatchSize) {
+    // Enqueue item.
+    NSDictionary *dict = [strongSelf dictionaryForTelemetryData:item];
+    [strongSelf appendDictionaryToEventBuffer:dict];
+    if (strongSelf.dataItemCount >= strongSelf.maxBatchSize) {
```

Comment:
You need an `@synchronized(self)` here as well to guard the access to the data item count.